worker/proposal_test.go

```diff
@@ -79,12 +79,14 @@ func TestLimiterDeadlock(t *testing.T) {
 	go func() {
 		now := time.Now()
 		for range ticker.C {
+			l.c.L.Lock()
```

Comment:
Why do you need the lock? For `l.iou`?

Reply:
There is concurrent write which is leading to the race condition. This fixes that.

Reply:
Where is the concurrent write?

Reply:
Got it.